src/main/kotlin/no/nb/mlt/wls/domain/model/Order.kt

```diff
@@ -91,11 +91,34 @@ data class Order(
         return status != Status.NOT_STARTED
     }
 
+    /**
+     * Makes a copy of the order with the updated fields
+     */
+    fun updateOrder(
+        itemIds: List<String>,
+        callbackUrl: String,
+        orderType: Type,
+        receiver: Receiver
+    ): Order {
+        throwIfInProgress()
+
+        return this.setOrderLines(itemIds)
+            .setCallbackUrl(callbackUrl)
+            .setOrderType(orderType)
+            .setReceiver(receiver)
+    }
+
+    /**
+     * Delete the order as long as it is possible
+     */
+    fun deleteOrder() {
+        throwIfInProgress()
+    }
+
     /**
      * Throws an exception if the order has been started or finished
-     * TODO - Make this private, and handle this in a more domain-driven way
      */
-    fun throwIfInProgress() {
+    private fun throwIfInProgress() {
```

Comment:
This is used in both delete and update, change the language of error messages to reflect that. i.e. "order can not be changed"